server/src/main/java/org/elasticsearch/cluster/routing/allocation/RoutingAllocation.java

```diff
@@ -81,7 +82,7 @@ public RoutingAllocation(
         SnapshotShardSizeInfo shardSizeInfo,
         long currentNanoTime
     ) {
-        this(deciders, null, clusterState, clusterInfo, shardSizeInfo, currentNanoTime);
+        this(deciders, null, clusterState, clusterInfo, shardSizeInfo, currentNanoTime, false, null);
```

Comment:
Hmm it's still `null` here. Could we have callers of this constructor specify their reason too?